### ISSUE TITLE
Buttons should not move when selected

### ISSUE THREAD

**Issue by wgx** [canonical-web-and-design/vanilla-framework#3713]
![115693906-a5df6100-a360-11eb-809c-39a081242575](https://user-images.githubusercontent.com/930099/115714556-23aa6900-a36f-11eb-8d07-885437cfa54c.gif)

In the example above, the visual feedback that "something is happening" is great - but the buttons should not change size or shape. 

It's too error prone and confusing to make clickable areas of UI move, suggest we re-approach this pattern to make it safer and more accessible.

**Comment by lyubomir-popov** [comment]
@amylily1011 do you remember where in maas we had something like this?

**Comment by amylily1011** [comment]
We have a loading state buttons in MAAS, but the button won't resize. 😃

**Comment by wgx** [comment]
☝🏼 This is what we want.

**Comment by lyubomir-popov** [comment]
@bartaz do we need more informatino here? button stays the size it was before the text disappears, loading icon centered within.

**Comment by bartaz** [comment]
No, I think it's enough. We can get to more details if needed when we implement it.

**Comment by bartaz** [comment]
Scope:
- investigate how it is implemented in React ActiveButton component
- try to replicate it in the example in Vanilla (add JS to keep the width of the button)
- if it is possible with just adding some JS add such example to Vanilla, if additional class names are needed propose how to amend button component to achieve that
